### PR TITLE
Allow precompilation

### PR DESCRIPTION
**Pull request by johnmyleswhite**
Title says it all.

### CODE REVIEW

File: src/Optim.jl [hunk 1]
```diff
@@ -1,4 +1,6 @@
 module Optim
+    VERSION >= v"0.4.0-dev+6521" && __precompile__(true)
```
Comment (mlubin) [inline src/Optim.jl]: Shouldn't this be outside ``module``?

Comment (johnmyleswhite) [inline src/Optim.jl]: I think you're right. Since this passed tests I didn't think about the fact that this is a file level property.

**Comment by johnmyleswhite** [comment]
As compilation is a file-level property, this was moved outside of the `module`.

**Comment by pkofod** [comment]
Any idea about the speed up of `using Optim`?

**Comment by johnmyleswhite** [comment]
It's there. The package loads fast enough for me now.

**Pull request merged**